### PR TITLE
Rename occurences of ~Then~ to ~Than~

### PR DESCRIPTION
**Pull request by erikpaperik**
Fixed a recurring typo where _then_ was used instead of _than_

### CODE REVIEW

**Comment by Goltergaul** [comment]
That was a pretty stupid naming mistake from my side :see_no_evil: 
However I'd like to avoid this renaming to be a breaking change and would propose that we keep all the old names as deprecated aliases for now

File: lib/definition/dsl/comparators.rb [hunk 1]
```diff
@@ -36,33 +36,33 @@ def NonEmptyString # rubocop:disable Style/MethodName
       end
 
       # Example:
-      # GreaterThen(5)
-      def GreaterThen(min_value) # rubocop:disable Style/MethodName
-        Types::Lambda.new("greater_then", context: { min_value: min_value }) do |value|
+      # GreaterThan(5)
+      def GreaterThan(min_value) # rubocop:disable Style/MethodName
```
Comment (Goltergaul) [inline lib/definition/dsl/comparators.rb:40]: pls add something like this to keep backwards compatibility:

```
alias GreaterThen GreaterThan
deprecate :GreaterThen
```

`deprecate` is coming with activesupport that we have included already FYI: https://apidock.com/rails/Module/deprecate

**Review by Goltergaul — CHANGES_REQUESTED** [review]

**Comment by TheWudu** [comment]
Is it already in use somewhere ? If not then i would prefer a breaking change ;-)

**Comment by Goltergaul** [comment]
@maf yes it is

**Review by TheWudu — APPROVED** [review]

**Comment by Goltergaul** [comment]
@erikpaperik do you have time to finish this pr?

**Review by Goltergaul — APPROVED** [review]

**Pull request merged**